helm/node-health-chart/Chart.yaml

```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
+appVersion: 0.1.0
+description: Kubernetes daemonset for dynamic and composable node health management.
+name: node-health-chart
+version: 0.1.0
```

Comment:
I am confused about all the versions here. There are three versions now and I do not know what they should be or how this should look like. Need feedback.

Reply:
This looks alright :+1: . To clarify:
- `apiVersion` is the helm api version.
- `appVersion` is the version of the (external) component which this chart is going to install (e.g. nginx-ingress-controller version). Since this project has no external component it can be left out, or just left at `0.1.0`.
- `version` is the version of the actual helm chart. This one is crucial for our deployment process and needs to be bumbed every time you change something.

Reply:
I dropped `appVersion`. Why do we have another version then in the values file with the image registry and repo?

Reply:
And where is the semver version then coming from? Do we need to add some publishing magic here?

Reply:
> Why do we have another version then in the values file with the image registry and repo
In the values file is just the templated image tag. Ideally this should be the same with the `appVersion` for external components, but helm gathers its release info from Chart.yaml, not from templated values.

Reply:
> And where is the semver version then coming from? Do we need to add some publishing magic here?

Yes, for `chart-operator` to be able to install the charts based on the version defined in the chartconfig (which cluster-operator creates), it needs the chart beeing published to quay.io to certain channels. This is already included into architect, like here:

https://github.com/giantswarm/kubernetes-nginx-ingress-controller/blob/ab6f1b9e8fc68af22e97293b6c0141615c64063c/.circleci/config.yml#L88

Reply:
I think you lost me. We dropped the `appVersion` here. I also do not understand what such an external component is. I feel like all these versions should actually be the same. Maintaining this version mess is rather ugly. Still wondering where the semver versions are coming from. Here we want to run the `node-health` in a daemonset. So how is the version tag created? It is usually the Git SHA and now it is a semver version. Where is it coming from and why is it different to the other projects?

Reply:
Helm has the idea of a
- **component version**, which depicts the version of the component you are going to deploy with this helm chart. `node-health` in this case, `nginx-ingress-controller` or `kube-state-metrics` in other cases.
- **chart version**, which depicts the version of the helm chart, (where you can make changes without changing the component)

The `version` field in `Chart.yaml` represents the **chart version** and we define it/ bump it ourselves. This will be automated in the future. Still this is the crucial version in the release publishing/release process. Chart-operator knows from this version, which chart to deploy. 

Also agree, all this versioning is still a bit confusing and should be simplified automated in the future. But I guess its a process.